src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -0,0 +1,130 @@
+
+case class ClusterRecord(id: Long,
+                         clusterName: String,
+                         googleId: UUID,
+                         googleProject: String,
+                         googleServiceAccount: String,
+                         googleBucket: String,
+                         operationName: String,
+                         status: String,
+                         hostIp: Option[String],
+                         createdDate: Timestamp,
+                         destroyedDate: Option[Timestamp])
+
+trait ClusterComponent extends LeoComponent {
+  this: LabelComponent =>
+
+  import profile.api._
+
+  class ClusterTable(tag: Tag) extends Table[ClusterRecord](tag, "CLUSTER") {
+    def id =                    column[Long]              ("id",                    O.PrimaryKey, O.AutoInc)
+    def clusterName =           column[String]            ("clusterName",           O.Length(254))
+    def googleId =              column[UUID]              ("googleId")
+    def googleProject =         column[String]            ("googleProject",         O.Length(254))
+    def googleServiceAccount =  column[String]            ("googleServiceAccount",  O.Length(254))
+    def googleBucket =          column[String]            ("googleBucket",          O.Length(254))
+    def operationName =         column[String]            ("operationName",         O.Length(254))
+    def status =                column[String]            ("status",                O.Length(254))
+    def hostIp =                column[Option[String]]    ("hostIp",                O.Length(254))
+    def createdDate =           column[Timestamp]         ("createdDate",           O.SqlType("TIMESTAMP(6)"))
+    def destroyedDate =         column[Option[Timestamp]] ("destroyedDate",         O.SqlType("TIMESTAMP(6)"))
+
+    def uniqueKey = index("IDX_CLUSTER_UNIQUE", (googleProject, clusterName), unique = true)
+
+    def * = (id, clusterName, googleId, googleProject, googleServiceAccount, googleBucket, operationName, status, hostIp, createdDate, destroyedDate) <> (ClusterRecord.tupled, ClusterRecord.unapply)
+  }
+
+  object clusterQuery extends TableQuery(new ClusterTable(_)) {
+
+    def save(cluster: Cluster): DBIO[Cluster] = {
+      (clusterQuery returning clusterQuery.map(_.id) += marshalCluster(cluster)) flatMap { clusterId =>
+        labelQuery.saveAll(clusterId, cluster.labels)
+      } map { _ => cluster }
+    }
+
+    def list(): DBIO[Seq[Cluster]] = {
+      clusterQuery.result flatMap { recs =>
+        DBIO.sequence(recs map unmarshalWithLabels)
+      }
+    }
+
+    // currently any string can be a GoogleProject
+    // but I'm planning to fix that soon
+    def getByName(project: GoogleProject, name: String): DBIO[Option[Cluster]] = {
+      clusterQuery.filter { _.googleProject === project }.filter { _.clusterName === name }.result flatMap { recs =>
+        DBIO.sequence(recs map unmarshalWithLabels) map { _.headOption }
```

Comment:
Not sure I like Rawls `uniqueResult` because it throws exceptions if there are too many elements -- I like using `headOption`, just want to enforce the uniqueness in the DB.